master/internal/user/postgres_users_intg_test.go

```diff
@@ -412,12 +424,156 @@ func addTestSession() (model.UserID, model.SessionID, string, error) {
 	return user.ID, session.ID, token, nil
 }
 
-func addTestUser(aug *model.AgentUserGroup) (*model.User, error) {
-	var user model.User
-	uid, err := Add(context.TODO(), &model.User{Username: uuid.NewString()}, aug)
+type opts func(*model.User)
```

Comment:
I like this ```opts``` pattern

in ```internal/db``` I tried a struct approach that was like ```mockExpParams {
  *SearcherType string
}```
but this mock approach is easier

besides a general question? Like do we need the named types ```opts``` or could the function just take a ```func(*model.User)```

Reply:
it could just take a func, yeah.